packages/mailgun/src/models/v1/emails/request/send-email-request/index.ts

```diff
@@ -1 +1,2 @@
 export type { SendEmailRequest } from './send-email-request';
+export { transformSendEmailRequestIntoApiRequestBody } from './send-email-request';
```

Comment:
All these function become public and outside of SDK internal usage scope.
This could cause namespace confusion/pollution.
Is there a way to reduce their scope/visibility to SDK internals only, and then reduce risk to have users with breaking change in case of internal changes in the future ?

Reply:
There is no risk of namespace confusion as all the exports inside the `models` package are exported through a namespace. 
Now, regarding the risk to expose the method externally, due to this barrel file the method is indeed exposed on the Mailgun namespace and can be used this way:
```
import { Mailgun } from '@sinch/sdk-core';
Mailgun.transformSendEmailRequestIntoApiRequestBody({ foo: 'bar' });
```

There are several way to circumvent the issue, more or less impactful and more or less working:
 - option 1: remove this export from the barrel file. The method is still exposed and a warning can be added to explicitly document this is an internal file that should not be used and can be renamed at any time. In order to use it, a SDK user would have to dig into the whole repository arborescence, such as
```
import {
  transformSendEmailRequestIntoApiRequestBody
} from '@sinch/mailgun/src/models/v1/emails/request/send-email-request';
```
 - option 2: remove this method from the model file and add it inline in the emails-api.ts file, along with the API call. This solves completely the issue but will bring some additional complexity to test it. Indeed, the test will have to be written in the same file, such as 
 ```
 if (process.env.NODE_ENV === 'test') {
  describe('transformSendEmailRequestIntoApiRequestBody', () => {
    test('should transform email request correctly', () => {
      const result = transformSendEmailRequestIntoApiRequestBody({ foo: 'bar' });
      expect(result.body).toBe(JSON.stringify({ foo: 'bar' }));
    });
  });
}
 ```
 On top of polluting the src file, this will increase the size of the final bundle as this code will be present in the NPM package.
 
  - option 3: extract the transform code in other files with a specific pattern such as `send-email-request.transform.ts`, write a build script that will inject the code of the transformers method in the right place in the emails-api.ts file and exclude the *.transform.ts files from being part of the NPM package. This way, the method is still testable and becomes private to the SDK user. This approach is the most conmplex and most costly to put in place.

Let's discuss about the 3 options during a standup to decide which one to choose.

Reply:
Team decision => let's go for option 1